Videomaker theme: Customize onboarding tour images and text

jgcaruso opened the pull request:
#### Proposed Changes

* Adjust the onboarding tour to have Video specific images and text when the Videomaker or Videomaker White themes are installed

**For Translators**
<img width="511" alt="Screen Shot 2022-10-24 at 1 31 05 PM" src="https://user-images.githubusercontent.com/4081020/197589185-c737a510-a2d1-492a-bb8e-8adc16c8af44.png">


#### Testing Instructions
**FYI:** This might not work great while testing onboarding unless you sandbox the new domain after it is created (**I THINK**) Testing instructions are based on just running a known site directly. We can deploy this separately if it makes sense to do so.

- apply diff to sandbox (for new assets) D89705-code
- sandbox `s0.wp.com`
- pull PR
- `cd apps/editing-toolkit`
- ensure your wpcom sandbox ssh alias is called `wpcom-sandbox` (**OR** edit it to your alias in `packages/calypso-apps-builder/index.js` **OR** figure out how the `remotePath` param works)
- `yarn dev --sync`
- visit a sandboxed site, ensure your theme is one of the `videomaker` variants (make sure to test both, since they are detected differently)
- visit the Full Site, Post or Page editor
- open the side menu (3 dots icon in upper right)
- Under Tools, select Welcome Guide
- Move through the cards, the 1st and 4th items should match the designs in Automattic/greenhouse#1301
- visit a sandboxed site on a different theme (or change your current theme)
- visit the editor again and launch the Welcome Guide
- the cards should appear as they do in prod

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

*

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

jgcaruso committed:
wip

jgcaruso committed:
update text for 4th card

jgcaruso committed:
added check for theme

roundhill commented on apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/tour-launch.tsx:
```diff
@@ -169,6 +171,11 @@ function WelcomeTour() {
 	};
 
+	// Theme isn't immediately available, so we prevent rendering so the content doesn't switch after it is presented, since some content is based on theme
+	if ( null === themeName ) {
```
Does this still show automatically after you onboard with a new site? Not sure what you mean by `immediately` :D

jgcaruso replied:
`immediately` just means on first render of the component. I was seeing the original coffee cup welcome image and then a second or 2 later the site info / theme request came back and populated the `themeName` variable and then the videomaker image popped in.

roundhill commented on apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/tour-steps.tsx:
```diff
@@ -54,8 +60,10 @@ function getTourAssets( key: string ): TourAsset | undefined {
+	isSiteEditor = false,
+	themeName: string | null = null
 ): WpcomStep[] {
+	const isVideoMaker = [ 'videomaker', 'videomaker white' ].indexOf( themeName ) >= 0;
```
I saw a cute (ES6?) solution once that read something like: 

```
[ 'videomaker', 'videomaker white' ].some( videomakerTheme => themeName.includes( videomakerTheme ) );
```

Yours works the same though ;)

tyxla replied:
Seems like this one would be even shorter, while keeping the same precision:

```suggestion
	const isVideoMaker = [ 'videomaker', 'videomaker white' ].includes( themeName );
```

jgcaruso replied:
Thanks, using `includes` looks much cleaner!

roundhill reviewed (approved):
Tested really well for me:

* Videomaker dark shows new tour ✅ 
* Videomaker white shows new tour ✅ 
* Other theme shows original tour ✅ 

Had a few non-blocking thoughts.

noahtallen commented:
Just a drive by comment about the testing instructions. There's actually an easier way to do this! Just run `install-plugin.sh editing-toolkit update/videomaker-update-tour-images` in your sandbox and the editing toolkit plugin will be updated to match the code on this branch (it also only takes a second or two to download everything). This way, you don't have to use the local dev environment just for testing a PR.

worldomonation commented:
Another drive-by comment, this time about rebase/merge workflows. [This P2 post](https://codep2.wordpress.com/2022/09/15/using-git-rebase-effectively/#comment-10717) may be useful to learn ways to avoid the review bombs!

jgcaruso committed:
replace `indexof` with `includes`

jgcaruso committed:
Add `ts-ignore` to bypass missing type information.

Upgrading the dependency doesn't seem to help, its possible the upstream
hasn't bee updated in a while and the function definition is just
missing from it.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7590993

Hi @jgcaruso, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Use the toolbar to change the appearance of a selected block. Try replacing a video!`

Thank you in advance!

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7590993).
